### PR TITLE
Delete old AllowanceTarget.t.sol foundry test contract, modify test f…

### PR DESCRIPTION
**Pull request by oneleo**
### Delete old AllowanceTarget.t.sol foundry test contract.
For better code maintenance.

### This PR also has following fixes for old/new file update conflicts:
- Modify test file name
- Add MIT SPDX-License-Identifier

### How to verify?

Please use following command to test:
```
% forge test -vvv --match-path "contracts/test/AllowanceTarget/*.t.sol" --force
```

The execution result is as follows:
<details>
  <summary>( Please click here to expand )</summary>
    <pre><code>

Running 2 tests for contracts/test/AllowanceTarget/Teardown.t.sol:TestAllowanceTargetTeardown
[PASS] testCannotTeardownByNotSpender() (gas: 13059)
[PASS] testTeardown() (gas: 14054)
Test result: ok. 2 passed; 0 failed; finished in 6.70ms

Running 3 tests for contracts/test/AllowanceTarget/SetupTest.t.sol:TestAllowanceTargetSetup
[PASS] testCannotConstructByZeroAddress() (gas: 41010)
[PASS] testConstructor() (gas: 7739)
[PASS] testSetupAllowance() (gas: 7761)
Test result: ok. 3 passed; 0 failed; finished in 6.17ms

Running 2 tests for contracts/test/AllowanceTarget/ExecuteCall.t.sol:TestAllowanceTargetExecuteCall
[PASS] testCannotExecuteByNotSpender() (gas: 15993)
[PASS] testExecuteCall() (gas: 832810)
Test result: ok. 2 passed; 0 failed; finished in 7.36ms

Running 3 tests for contracts/test/AllowanceTarget/CompleteSetSpender.t.sol:TestAllowanceTargetCompleteSetSpender
[PASS] testCannotCompleteBeforeSet() (gas: 10620)
[PASS] testCannotCompleteTooEarly() (gas: 60817)
[PASS] testCompleteSetSpender() (gas: 48614)
Test result: ok. 3 passed; 0 failed; finished in 7.51ms

Running 4 tests for contracts/test/AllowanceTarget/SetSpenderWithTimelock.t.sol:TestAllowanceTargetSetSpenderWithTimelock
[PASS] testCannotSetByInvalidAddress() (gas: 15503)
[PASS] testCannotSetByRandomEOA() (gas: 15318)
[PASS] testCannotSetInProgress() (gas: 141333)
[PASS] testSetSpenderWithTimelock() (gas: 58831)
Test result: ok. 4 passed; 0 failed; finished in 7.51ms
    </code></pre>
</details>

Thank you very much!

### CODE REVIEW

File: contracts/test/AllowanceTarget/SetupTest.t.sol [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+// SPDX-License-Identifier: MIT
 pragma solidity 0.7.6;
 
 import "contracts-test/AllowanceTarget/Setup.t.sol";
 
-contract TestAllowanceTargetConstructor is TestAllowanceTarget {
+contract TestAllowanceTargetSetup is TestAllowanceTarget {
+    function testAllowanceTargetSetup() public {
+        assertEq(allowanceTarget.spender(), address(this));
+    }
```
Comment (NIC619) [inline contracts/test/AllowanceTarget/SetupTest.t.sol:9]: This is also the test of constructor? Maybe the `Test setup: constructor` comment  should be put before this test?

Comment (oneleo) [reply contracts/test/AllowanceTarget/SetupTest.t.sol:9]: Yes, it is also similar to test which creates a new contract in Setup.t.sol...
Or should we remove the `Test setup: constructor` comment to reduce code?

Comment (NIC619) [reply contracts/test/AllowanceTarget/SetupTest.t.sol:9]: Yeah I think now that tests are separated into files, the comment can be removed.

Comment (oneleo) [reply contracts/test/AllowanceTarget/SetupTest.t.sol:9]: OK! Thank you!

**Review by pilagod — APPROVED** [review]

**Review by charlesjhongc — APPROVED** [review]

**Review by NIC619 — APPROVED** [review]

**Pull request merged**